aws-auth-cognito/src/androidTest/java/com/amplifyframework/auth/cognito/AuthCanaryTest.kt

```diff
@@ -0,0 +1,688 @@
+                AuthUserAttributeKey.email(),
+                "344299",
+                {
+                    Log.i(TAG, "Confirmed user attribute with correct code.")
+                    fail()
+                },
+                {
+                    Log.e(TAG, "Failed to confirm user attribute. Bad code?", it)
+                    latch.countDown()
+                }
+            )
+        } catch (e: Exception) {
+            fail(e.toString())
+        }
+        assertTrue(latch.await(TIMEOUT_S, TimeUnit.SECONDS))
+    }
+
+    @Test
+    fun resendUserAttributeConfirmationCode() {
+        signInBeforeTest(username, password)
+        val latch = CountDownLatch(1)
+        try {
+            Amplify.Auth.resendUserAttributeConfirmationCode(
+                AuthUserAttributeKey.email(),
+                {
+                    Log.i(TAG, "Code was sent again: $it")
+                    latch.countDown()
+                },
+                {
+                    Log.e(TAG, "Failed to resend code", it)
+                    fail()
+                }
+            )
+        } catch (e: Exception) {
+            fail(e.toString())
+        }
+        assertTrue(latch.await(TIMEOUT_S, TimeUnit.SECONDS))
+    }
+
+    @Test
+    fun getCurrentUser() {
+        signInBeforeTest(username, password)
+        val latch = CountDownLatch(1)
+        try {
+            Amplify.Auth.getCurrentUser(
+                {
+                    Log.i(TAG, "Current user details are: $it")
+                    latch.countDown()
+                },
+                {
+                    Log.e(TAG, "getCurrentUser failed with an exception: $it")
+                    fail()
+                }
+            )
+        } catch (e: Exception) {
+            fail(e.toString())
+        }
+        assertTrue(latch.await(TIMEOUT_S, TimeUnit.SECONDS))
+    }
+
+    @Test
+    fun signOut() {
+        signInBeforeTest(username, password)
+        val latch = CountDownLatch(1)
+        try {
+            Amplify.Auth.signOut {
+                Log.i(TAG, "Signed out successfully")
+                latch.countDown()
+            }
+        } catch (e: Exception) {
+            fail(e.toString())
+        }
+        assertTrue(latch.await(TIMEOUT_S, TimeUnit.SECONDS))
+    }
+
+    @Test
+    fun globalSignOut() {
+        signInBeforeTest(username, password)
+        val latch = CountDownLatch(1)
+        val options = AuthSignOutOptions.builder()
+            .globalSignOut(true)
+            .build()
+        try {
+            Amplify.Auth.signOut(options) {
+                Log.i(TAG, "Signed out successfully")
+                latch.countDown()
+            }
+        } catch (e: Exception) {
+            fail(e.toString())
+        }
+        assertTrue(latch.await(TIMEOUT_S, TimeUnit.SECONDS))
+    }
+
+    @Test
+    @Ignore("Need to sign up a tempUser and signIn, which requires confirmation.")
+    fun deleteUser() {
```

Comment:
Can we use admin API to delete the user?

We are signing up new users before a lot of tests. We should be deleting them after the test is complete.